packages/react/src/components/Table/StatefulTable.jsx

```diff
@@ -347,7 +347,7 @@ const StatefulTable = ({ data: initialData, expandedData, ...other }) => {
   const filteredTotalItems =
     filteredCount && filteredCount !== currentlyLoadedDataCount
       ? filteredCount
-      : totalItems || currentlyLoadedDataCount;
+      : view?.pagination?.totalItems || totalItems || currentlyLoadedDataCount;
```

Comment:
I don't understand this. Isn't the `totalItems` is destructed from `view.?pagination?.totalItems` already?

Reply:
Yeah, it took me a little bit to figure this one out, too. The [defaultProps function in table](https://github.com/carbon-design-system/carbon-addons-iot-react/blob/next/packages/react/src/components/Table/Table.jsx#L322) passes the length of the data from the initial data as the length of the totalItems, so the initial data passed always overrides the length of the loaded (and in this case changed) data. Thus, the `totalItems` destructed from the merging of that data is always 100 in this case. This may not be the best way to fix it, IDK. But that's the situation.

Reply:
Ahh, I see. If this is the setup needed, can we rename `totalItems` to `initialTotalItems` to clarify the difference?

Reply:
done.